### ISSUE TITLE
WPF - Mouse cursor not changing (cursor is none)

### ISSUE THREAD

**Issue by phkunz** [cefsharp/CefSharp#4021]
- **What version of the product are you using?**
     v98.1.210

- **What architecture x86 or x64?**
    x64
	
- **What version of .Net?**
    tested with .Net 5.0 and .Net Framework 4.7.2 
    
- **On what operating system?**
    Win10

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
    Only WPF has the issue. Issue does not appear in WinForms
    
- **What steps will reproduce the problem?**
    - Start CefSharp.Wpf.Example project in debug configuration.
    - Press middle button in the application to enter "AutoScroll" mode

- **What is the expected output? What do you see instead?**
    - It is expected to see an icon with arrows depending on the current operation.
    - Instead no icon is shown.

**Comment by amaitland** [comment]
Please test with the CEF Sample application as detailed in the bug report temple. 

https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#bug-report

I expect this is an upstream issue which needs to be fixed in CEF.

**Comment by amaitland** [comment]
Probably related to https://bitbucket.org/chromiumembedded/cef/issues/3174/panning-mouse-cursors-not-showing

**Comment by amaitland** [comment]
Quick test confirms that this reproduces with the `CEF Sample Application(cefclient)`. This should be fixed `CEF`, please create an issue on the [CEF Issue Tracker](https://bitbucket.org/chromiumembedded/cef/issues?status=new&status=open) and post a link back here for reference.

You should be able to implement a temporary workaround using something like the following (doesn't cover all cases, gives the general idea though): 

```c#
public class ChromiumWebBrowserEx : ChromiumWebBrowser
{

	protected override void OnCursorChange(IntPtr handle, CefSharp.Enums.CursorType type, CursorInfo customCursorInfo)
	{
		switch (type)
		{
			case CefSharp.Enums.CursorType.MiddlePanningVertical:
			{
				Dispatcher.InvokeAsync(() =>
				{
					Cursor = Cursors.ScrollNS;
				});

				break;
			}
			case CefSharp.Enums.CursorType.NorthPanning:
			{
				Dispatcher.InvokeAsync(() =>
				{
					Cursor = Cursors.ScrollN;
				});

				break;
			}
			case CefSharp.Enums.CursorType.SouthPanning:
			{
				Dispatcher.InvokeAsync(() =>
				{
					Cursor = Cursors.ScrollS;
				});

				break;
			}
			default:
			{
				base.OnCursorChange(handle, type, customCursorInfo);

				break;
			}
		}

	}
}
```

Closing as `upstream`.

**Comment by amaitland** [comment]
If it doesn't look like the issue will be fixed in `CEF` in the next version or two then we can potentially look at adding the workaround, will wait to see the response.

**Comment by phkunz** [comment]
Thank You for the fast reply and the workaround. I forgot to mention that  I tested it with the sample application myself using https://cef-builds.spotifycdn.com/cef_binary_98.1.21%2Bg9782362%2Bchromium-98.0.4758.102_windows64_client.tar.bz2 and this worked for me (see attached screenshot).
<img width="696" alt="Screenshot 2022-03-01 090340" src="https://user-images.githubusercontent.com/38318227/156129219-3ff3655f-056b-441e-8f5f-9341c0ad43b6.png">

**Comment by amaitland** [comment]
> I forgot to mention that I tested it with the sample application myself using https://cef-builds.spotifycdn.com/cef_binary_98.1.21%2Bg9782362%2Bchromium-98.0.4758.102_windows64_client.tar.bz2 and this worked for me (see attached screenshot).

Did you test with the command line args listed in the [Bug Report](https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#bug-report)?

```
cefclient.exe --multi-threaded-message-loop --no-sandbox --off-screen-rendering-enabled --enable-gpu
```

It's **important*** you test with the args listed above. In particular `--off-screen-rendering-enabled`.

**Comment by phkunz** [comment]
With `cefclient.exe --multi-threaded-message-loop --no-sandbox --off-screen-rendering-enabled --enable-gpu` I can reproduce the issue in CEF. Sorry for not reading the instruction properly. I will post the issue upstream.

**Comment by amaitland** [comment]
> With `cefclient.exe --multi-threaded-message-loop --no-sandbox --off-screen-rendering-enabled --enable-gpu` I can reproduce the issue in CEF.

`cefclient.exe --off-screen-rendering-enabled` should be sufficent to reproduce the problem and will likely simplify your bug report on the `CEF Issue tracker`. The `WPF` version uses `offscreen rendering`.

Another alternative is https://github.com/cefsharp/CefSharp.Wpf.HwndHost#cefsharpwpfhwndhost which is the equivalent of hosting the `WinForms` version in `WPF` (airspace issues apply).

**Comment by phkunz** [comment]
I created an upstream issue
https://bitbucket.org/chromiumembedded/cef/issues/3270/osr-pan-icon-missing-when-pressing-the

**Comment by amaitland** [comment]
Thanks

**Comment by amaitland** [comment]
If the upstream issue isn't resolved shortly we'll need to add a working. 

Reopening.

**Comment by amaitland** [comment]
I've implemented a workaround which is now available in the [100.0.120-pre](https://github.com/cefsharp/CefSharp/issues/4053#issuecomment-1088135006) release. 

Anyone interested please test and report back.

**Comment by mayphi** [comment]
> I've implemented a workaround which is now available in the [100.0.120-pre](https://github.com/cefsharp/CefSharp/issues/4053#issuecomment-1088135006) release.
> 
> Anyone interested please test and report back.

Works fine on Win10 x64 with .Net 5
